app/src/actions/addStake.ts

```diff
@@ -48,23 +62,84 @@ export const addStake = async (
   if (!tokenOwnerRecord) {
     // if there is no prior token owner record
     // create one so that the owner can cast votes
+    let createStakerGovRecord: TransactionInstruction[] = [];
     await withCreateTokenOwnerRecord(
-      instructions,
+      createStakerGovRecord,
       GOVERNANCE_PROGRAM_ID,
       stakePool.stakePool.governanceRealm,
       owner,
       stakePool.stakePool.tokenMint,
       owner
     );
+
+    allTxs.push({
+      tx: new Transaction().add(...createStakerGovRecord),
+      signers: []
+    });
+  } else if (tokenOwnerRecord.account.unrelinquishedVotesCount > 0) {
+    // Rescind, then re-cast votes on any currently voting proposals
+    // In order to update vote weight
+
+    const voteRecords = await getUnrelinquishedVoteRecords(
+      connection,
+      programId,
+      tokenOwnerRecord.account.governingTokenOwner
+    );
+    for (const i in voteRecords) {
+      let proposal = proposals[voteRecords[i].account.proposal.toString()];
+      if (!proposal) {
+        continue;
+      }
+      const relinquishIxs: TransactionInstruction[] = [];
+      withRelinquishVote(
+        relinquishIxs,
+        programId,
+        proposal.account.governance,
+        proposal.pubkey,
+        tokenOwnerRecord.pubkey,
+        proposal.account.governingTokenMint,
+        voteRecords[i].pubkey,
+        tokenOwnerRecord.account.governingTokenOwner,
+        walletPubkey
+      );
+      allTxs.push({
+        tx: new Transaction().add(...relinquishIxs),
+        signers: []
+      });
+    }
+
+    for (const i in voteRecords) {
+      let proposal = proposals[voteRecords[i].account.proposal.toString()];
+      if (!proposal || proposal.account.hasVoteTimeEnded(governance.account)) {
+        continue;
+      }
+      if (voteRecords[i].account.vote && stakeAccount) {
+        const recastIxs: TransactionInstruction[] = [];
+        await withCastVote(
+          recastIxs,
+          programId,
+          programVersion,
+          stakePool.stakePool.governanceRealm,
+          proposal.account.governance,
+          proposal.pubkey,
+          proposal.account.tokenOwnerRecord,
+          tokenOwnerRecord.pubkey,
+          walletPubkey,
+          proposal.account.governingTokenMint,
+          voteRecords[i].account.vote!,
```

Comment:
You can pull `voteRecords[i].account.vote` into it's own variable next to `const proposal` and check if it's undefined
```suggestion
          voteRecords[i].account.vote,
```